zebra-chain/src/mmr.rs

```diff
@@ -0,0 +1,43 @@
+//! Merkle mountain range structure that contains information about
+//! the block history as specified in ZIP-221.
+
+use std::borrow::Borrow;
+
+use crate::block::{Block, ChainHistoryMmrRootHash};
+
+/// Merkle mountain range structure.
+pub struct MerkleMountainRange {
+    // TODO
+}
+
+impl MerkleMountainRange {
+    pub fn push(&mut self, block: &Block) {
+        // TODO: zcash_history::Tree::append_leaf receives an Arc<Block>. Should
+        // change that to match, or create an Arc to pass to it?
```

Comment:
In the caller:
- `prepared.block.as_ref()` currently gets a `&Block`
- `prepared.block.clone()` will get an `Arc<Block>`

If something else in the function needs a `&Block`, just use `block.as_ref()` for that specific code.

Reply:
Got it, thanks!